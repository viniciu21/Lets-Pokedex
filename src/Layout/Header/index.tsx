import React from "react";
import {
	AppBar,
	Avatar,
	Button,
	IconButton,
	Toolbar,
	Typography,
	Badge,
} from "@material-ui/core";
import { Link } from "react-router-dom";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import pokebola from "../../assets/pokebola.jpg";
import NotificationsIcon from "@material-ui/icons/Notifications";
import MailIcon from "@material-ui/icons/Mail";
import AccountCircle from "@material-ui/icons/AccountCircle";

const styles = makeStyles((theme: Theme) =>
	createStyles({
		root: {
			flexGrow: 1,
			background: "#673ab7",
			width: "100%",
		},
		title: {
			[theme.breakpoints.up("sm")]: {
				display: "block",
			},
		},
		avatar: {
			// marginLeft: theme.spacing(1),
		},
		routes: {
			// marginLeft: theme.spacing(15),
		},
	})
);

const Header = () => {
	const classes = styles();

	return (
		<div className={classes.root}>
			<AppBar position="relative">
				<Toolbar>
					<Typography className={classes.title} variant={"h6"}>
						LET'S POKEDEX
					</Typography>

					<Avatar className={classes.avatar} src={pokebola} />

					<div className={classes.root} />

					<Link to="/" style={{ textDecoration: "none", color: "white" }}>
						<Button color="inherit" className={classes.routes}>
							All pokemons
						</Button>
					</Link>

					<div className={classes.root} />

					<Link to="/stared" style={{ textDecoration: "none", color: "white" }}>
						<Button color="inherit" className={classes.routes}>
							Stared Pokemons
						</Button>
					</Link>

					<div className={classes.root} />

					<IconButton edge="end" aria-label="notifications" color="inherit">
						<Badge badgeContent={4} color="secondary">
							<NotificationsIcon />
						</Badge>
					</IconButton>

					<IconButton
						className={classes.avatar}
						aria-label="notifications"
						color="inherit"
					>
						<Badge badgeContent={4} color="secondary">
							<MailIcon />
						</Badge>
					</IconButton>

					<IconButton color="inherit" className={classes.avatar}>
						<AccountCircle />
					</IconButton>
				</Toolbar>
			</AppBar>
		</div>
	);
};

export default Header;
