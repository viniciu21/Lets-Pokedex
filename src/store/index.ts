import { createStore, applyMiddleware, Store } from "redux";
import createSagaMiddleware from "redux-saga";
import { PokedexState } from "./ducks/pokedex/types";

import rootSaga from "./ducks/rootSaga";
import rootReducer from "./ducks/rootReducer";

export interface ApplicationState {
	pokedexReducer: PokedexState;
}

const sagaMiddleware = createSagaMiddleware();
const store: Store<ApplicationState> = createStore(
	rootReducer,
	applyMiddleware(sagaMiddleware)
);
sagaMiddleware.run(rootSaga);

export default store;
