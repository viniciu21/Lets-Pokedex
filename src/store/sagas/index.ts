import { all, takeLatest } from "redux-saga/effects";
import { PokedexTypes } from "../types/pokedex";
import { getInformationPokemons } from "./pokedex";

export default function* rootSaga() {
	return yield all([
		takeLatest(PokedexTypes.getDataPokemonsStart, getInformationPokemons),
	]);
}
