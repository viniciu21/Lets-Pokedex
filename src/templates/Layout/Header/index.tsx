import React from "react";
import {
	AppBar,
	Avatar,
	Button,
	IconButton,
	Toolbar,
	Typography,
	Badge,
} from "@material-ui/core";
import { Link } from "react-router-dom";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import pokebola from "../../../assets/pokebola.jpg";
import NotificationsIcon from "@material-ui/icons/Notifications";
import MailIcon from "@material-ui/icons/Mail";
import AccountCircle from "@material-ui/icons/AccountCircle";

const styles = makeStyles((theme: Theme) =>
	createStyles({
		root: {
			flexGrow: 1,
		},
		title: {
			[theme.breakpoints.up("sm")]: {
				display: "block",
			},
		},
		avatar: {
			marginLeft: theme.spacing(1),
		},
		routes: {
			marginLeft: theme.spacing(15),
		},
	})
);

const Header = () => {
	const classes = styles();

	return (
		<div className={classes.root}>
			<AppBar position="static">
				<Toolbar>
					<Typography className={classes.title} variant={"h6"}>
						LET'S POKEDEX
					</Typography>

					<Avatar className={classes.avatar} src={pokebola} />

					<Link to="/" style={{ textDecoration: "none", color: "white" }}>
						<Button color="inherit" className={classes.routes}>
							Todos os pokemons
						</Button>
					</Link>

					<Link to="/stared" style={{ textDecoration: "none", color: "white" }}>
						<Button color="inherit" className={classes.routes}>
							Pokemons favoritos
						</Button>
					</Link>

					<div className={classes.root} />

					<IconButton edge="end" aria-label="notifications" color="inherit">
						<Badge badgeContent={4} color="secondary">
							<NotificationsIcon />
						</Badge>
					</IconButton>

					<IconButton
						className={classes.avatar}
						// edge="end"
						aria-label="notifications"
						color="inherit"
					>
						<Badge badgeContent={4} color="secondary">
							<MailIcon />
						</Badge>
					</IconButton>

					<IconButton color="inherit" className={classes.avatar}>
						<AccountCircle />
					</IconButton>
				</Toolbar>
			</AppBar>
		</div>
	);
};

export default Header;
