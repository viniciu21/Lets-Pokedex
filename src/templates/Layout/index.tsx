import React from "react";
import { CssBaseline } from "@material-ui/core";
import Header from "./Header";
import { BrowserRouter } from "react-router-dom";
import Routes from "../../routes";

const Layout = () => {
	return (
		<div style={{ display: "flex", flexDirection: "column" }}>
			<BrowserRouter>
				<CssBaseline />
				<Header />
				<Routes />
			</BrowserRouter>
		</div>
	);
};

export default Layout;
