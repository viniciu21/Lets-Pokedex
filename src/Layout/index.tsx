import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getDataPokemonsStart } from "../store/ducks/pokedex/actions";
import { CssBaseline } from "@material-ui/core";
import Header from "./Header";
import { BrowserRouter } from "react-router-dom";
import Routes from "../routes";
import {ApplicationState} from '../store';

const Layout = () => {

	const {pages} = useSelector<ApplicationState, ApplicationState["pokedexReducer"]>((state) => state.pokedexReducer)
	const dispatch = useDispatch();
	useEffect(() => {
		dispatch(getDataPokemonsStart(pages));
	}, [pages, dispatch])
	return (
		<div
			style={{
				display: "flex",
				flexDirection: "column",
			}}
		>
			<BrowserRouter>
				<CssBaseline />
				<Header />
				<Routes />
			</BrowserRouter>
		</div>
	);
};

export default Layout;
