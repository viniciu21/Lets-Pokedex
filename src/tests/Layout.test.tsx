import React from 'react';
import { shallow } from 'enzyme';
import Layout from '../Layout';
import { Provider } from 'react-redux';
import store from '../store';

it("Shoud be possibel render a component layout", ()=> {
	const component = shallow(
		<Provider store={store}>
			<Layout ></Layout>
			</Provider>	);
	expect(component).toMatchSnapshot();
})
