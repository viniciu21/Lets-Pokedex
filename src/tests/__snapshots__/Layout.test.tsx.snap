// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoud be possibel render a component layout 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Layout />
</ContextProvider>
`;

exports[`renders the component Shoud be possibel render a component layout 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Layout />
</ContextProvider>
`;
